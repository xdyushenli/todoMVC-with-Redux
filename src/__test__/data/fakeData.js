const activeItem = {
    isActive: true,
    id: 1,
    text: 'active item'
}

const completedItem = {
    isActive: false,
    id: 1,
    text: 'completed item'
}

const todoList = [
    {
        isActive: false,
        id: 1,
        text: 'completed item'
    },
    {
        isActive: true,
        id: 2,
        text: 'active item'
    }
]

export {activeItem, completedItem, todoList}