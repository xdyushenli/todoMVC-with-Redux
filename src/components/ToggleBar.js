import React from 'react';
import '../index.css';

export default class ToggleBar extends React.Component{
    constructor(props) {
        super(props);
    }

    handleSwitchAll(e) {
        this.props.switchAll(e);
    }

    handleSwitchActive(e) {
        this.props.switchActive(e);
    }

    handleSwitchCompleted(e) {
        this.props.switchCompleted(e);
    }

    handleClearCompleted(e) {
        this.props.clearCompleted(e);
    }

    render() {
        let itemsNum = 0;
        let todoList = this.props.todoList;
        let showMode = this.props.showMode;

        todoList.map((item) => {
            if(item.isActive) itemsNum += 1;
        })

        return (
            <div id='toggle-bar'>
                <p className='left-items'>{itemsNum} items left</p>
                <div className='toggle-group'>
                    <button className={showMode === 'all' ? 'all-items active' : 'all-items'}
                            onClick={(e) => this.handleSwitchAll(e)}
                    >All</button>
                    <button className={showMode === 'active' ? 'active-items active' : 'active-items'}
                            onClick={(e) => this.handleSwitchActive(e)}                    
                    >Active</button>
                    <button className={showMode === 'completed' ? 'completed-items active' : 'completed-items'}
                            onClick={(e) => this.handleSwitchCompleted(e)}                    
                    >Completed</button>
                </div>
                <button className='clear-completed'
                        onClick={(e) => this.handleClearCompleted(e)}
                >Clear completed</button>
            </div>
        )
    }
}