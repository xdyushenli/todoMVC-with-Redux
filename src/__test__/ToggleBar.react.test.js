import React from 'react';
import ToggleBar from '../components/ToggleBar';
import renderer from 'react-test-renderer';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {todoList} from './data/fakeData';

Enzyme.configure({adapter: new Adapter()});

it('render all toggle', () => {
    const tree = renderer.create
        (
            <ToggleBar
                todoList={todoList}
                showMode='all'
            ></ToggleBar>
        ).toJSON();

    expect(tree).toMatchSnapshot();
})

it('render active toggle', () => {
    const tree = renderer.create
        (
            <ToggleBar
                todoList={todoList}
                showMode='active'
            ></ToggleBar>
        ).toJSON();

    expect(tree).toMatchSnapshot();
})

it('render completed toggle', () => {
    const tree = renderer.create
        (
            <ToggleBar
                todoList={todoList}
                showMode='completed'
            ></ToggleBar>
        ).toJSON();

    expect(tree).toMatchSnapshot();
})