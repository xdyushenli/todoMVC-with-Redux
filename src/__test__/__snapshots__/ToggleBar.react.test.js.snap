// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render active toggle 1`] = `
<div
  id="toggle-bar"
>
  <p
    className="left-items"
  >
    1
     items left
  </p>
  <div
    className="toggle-group"
  >
    <button
      className="all-items"
      onClick={[Function]}
    >
      All
    </button>
    <button
      className="active-items active"
      onClick={[Function]}
    >
      Active
    </button>
    <button
      className="completed-items"
      onClick={[Function]}
    >
      Completed
    </button>
  </div>
  <button
    className="clear-completed"
    onClick={[Function]}
  >
    Clear completed
  </button>
</div>
`;

exports[`render all toggle 1`] = `
<div
  id="toggle-bar"
>
  <p
    className="left-items"
  >
    1
     items left
  </p>
  <div
    className="toggle-group"
  >
    <button
      className="all-items active"
      onClick={[Function]}
    >
      All
    </button>
    <button
      className="active-items"
      onClick={[Function]}
    >
      Active
    </button>
    <button
      className="completed-items"
      onClick={[Function]}
    >
      Completed
    </button>
  </div>
  <button
    className="clear-completed"
    onClick={[Function]}
  >
    Clear completed
  </button>
</div>
`;

exports[`render completed toggle 1`] = `
<div
  id="toggle-bar"
>
  <p
    className="left-items"
  >
    1
     items left
  </p>
  <div
    className="toggle-group"
  >
    <button
      className="all-items"
      onClick={[Function]}
    >
      All
    </button>
    <button
      className="active-items"
      onClick={[Function]}
    >
      Active
    </button>
    <button
      className="completed-items active"
      onClick={[Function]}
    >
      Completed
    </button>
  </div>
  <button
    className="clear-completed"
    onClick={[Function]}
  >
    Clear completed
  </button>
</div>
`;
